dial_test.go

```diff
@@ -439,39 +431,18 @@ func TestDialBackoffClears(t *testing.T) {
 	if duration > 2*transport.DialTimeout*DialAttempts {
 		t.Error("> 2*transport.DialTimeout * DialAttempts not being respected", duration, 2*transport.DialTimeout*DialAttempts)
 	}
-
-	if !s1.Backoff().Backoff(s2.LocalPeer(), s2bad) {
-		t.Error("s2 should now be on backoff")
-	} else {
-		t.Log("correctly added to backoff")
-	}
+	require.True(t, s1.Backoff().Backoff(s2.LocalPeer(), s2bad), "s2 should now be on backoff")
 
 	// phase 2 -- add the working address. dial should succeed.
-	ifaceAddrs1, err := swarms[1].InterfaceListenAddresses()
-	if err != nil {
-		t.Fatal(err)
-	}
+	ifaceAddrs1, err := s2.InterfaceListenAddresses()
+	require.NoError(t, err)
 	s1.Peerstore().AddAddrs(s2.LocalPeer(), ifaceAddrs1, peerstore.PermanentAddrTTL)
 
-	if c, err := s1.DialPeer(ctx, s2.LocalPeer()); err == nil {
-		c.Close()
-		t.Log("backoffs are per address, not peer")
-	}
-
-	time.Sleep(BackoffBase)
```

Comment:
I'm not sure why were sleeping here. Removing this doesn't change anything about this test, so maybe the test case was broken in the first place?

Reply:
to clear thebackofd internally i believe

Reply:
but indeed doesnt seem to matter.